### PR TITLE
Broken sphinx install link and instructions #191

### LINKED ISSUE #191: broken sphinx install link / instructions
The "Installing Sphinx"  sub section of "Contributing to Rockstor documentation"(contribute_documentation.html) has a broken link:

http://www.sphinx-doc.org/latest/install.html

this could be replaced with:

http://www.sphinx-doc.org/en/master/usage/installation.html

and to add robustness we could also link to Sphinx's home page by changing our existing "Sphinx official documentation" text into a link to:

http://www.sphinx-doc.org/en/master/#

and rather than the existing "Follow the documentation here(link)" we cold have have, "Details of how to install Sphinx is covered in their **Get it**(link) subsection." Which also removed the questionable practice of linking the 'here' word.

### PR DESCRIPTION
**Pull request by Dbzs**
Updated links, added links to the extensions, included information on youtube extension problems and install instructions for Python 2 and 3.

Closes #191 

Successful make html and functionality check
Running Sphinx v1.8.4
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 1 source files that are out of date
updating environment: [] 0 added, 1 changed, 0 removed
reading sources... [100%] contribute_documentation
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index
generating indices... genindex
writing additional pages... search
copying static files... WARNING: html_static_path entry 'C:\\Users\\Del\\Documen
ts\\GitHub\\rockstor-doc\\_static' does not exist
done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 1 warning.

The HTML pages are in _build\html.

Build finished. The HTML pages are in _build/html.

### CODE REVIEW

**Comment by phillxnet** [comment]
@Dbzs Thanks for yet another doc pull request. And well done getting to the Sphinx stage. I've had a quick look and it all looks good, but can't take a proper look just now unfortunately.

Nice addition re the "Fixes # number of issue" along with GitHub references; which is ironic given your own use "Closes Issues" in this request. :)
If you could edit that to either a "Closes #  " ie no 'issue' text , or one of the other keywords your link indicated and we should get an auto close effect. Now worries really but given you are establishing a bit of a run here I'm just trying to get things lined up nice as we go.
Some nice additions here, shame we have that outstanding strangeness with Python3 but at least we will have that in the docs for now.

Will review soon, if others don't beat me to it of course.

**Comment by phillxnet** [comment]
@Dbzs Thanks for the "Closes ..." edit. Much appreciated.

File: contribute_documentation.rst [hunk 1]
```diff
@@ -147,6 +171,13 @@ following command::
 	[you@your_laptop rockstor-doc] git push origin your_branch_name
 
 When you finish work for the issue and are ready to submit, create a pull
-request by clicking on the “pull request” button on github. This notifies the
+request by clicking on the **pull request** button on github. This notifies the
 maintainers of your changes. As a best practice only open one pull request per
 issue containing all relevant changes.
+
+When you are ready open the pull request please follow these 2 tips to expedite the review.
```
Comment (FroggyFlox) [inline contribute_documentation.rst]:
```suggestion
When you are ready, open the pull request and please follow these 2 tips to expedite the review:
```

File: contribute_documentation.rst [hunk 2]
```diff
@@ -147,6 +171,13 @@ following command::
...
+
+    * When you've finished your edit, run the Sphinx *make html* command,and paste the output of the command into the issue discussion string to help speed up the review.  After you've generated the html, you can use the webserver detailed above to check the functionality of your work prior to your pull request
```
Comment (FroggyFlox) [inline contribute_documentation.rst]:
```suggestion
    * When you've finished your edit, run the Sphinx *make html* command, and paste its output into the pull request discussion string to help speed up the review.  After you've generated the html, you can use the webserver detailed above to check the functionality of your work prior to your pull request
```

File: contribute_documentation.rst [hunk 3]
```diff
@@ -147,6 +171,13 @@ following command::
...
+
+
+    * When you make a pull request, if you add a "Fixes #number-of-issue" on it's own line then when it gets merged it auto closes that issue. Just a nice to have and also provides a link to the relevant issue. See https://help.github.com/articles/closing-issues-using-keywords/ for details.
```
Comment (FroggyFlox) [inline contribute_documentation.rst]:
```suggestion
    * When you make a pull request, adding a "Fixes #number-of-issue" on its own line will automatically close the related issue when it gets merged. Just a nice thing to have and also provides a link to the relevant issue. See https://help.github.com/articles/closing-issues-using-keywords/ for details.
```

**Comment by Dbzs** [comment]
@FroggyFlox thanks for the suggestions, accepted them all.

**Comment by phillxnet** [comment]
@Dbzs Thanks for the doc addition. Nice addition. Should help others contribute here which is great.
Finally getting to merge this.
@FroggyFlox  Thanks for the review. Much appreciated.

**Pull request merged**